include/aws/common/mutex.h

```diff
@@ -0,0 +1,71 @@
+#ifndef AWS_COMMON_MUTEX_H_
+#define AWS_COMMON_MUTEX_H_
+
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/common.h>
+#ifdef _WIN32
+#include <Windows.h>
+#else
+#include <pthread.h>
+#endif
+
+struct aws_mutex {
+    struct aws_allocator *allocator;
```

Comment:
Is this needed?

Reply:
no, a long time ago we discussed keeping it there for long term api compatibility.

Reply:
but at the moment, thread and mutex don't need allocators